lib/kdump_utils.pm

```diff
@@ -171,7 +177,11 @@ sub activate_kdump {
     }
     send_key('alt-o');
     if ($expect_restart_info == 1) {
-        assert_screen('yast2-kdump-restart-info', 200);
+        my @tags = qw(yast2-kdump-restart-info os-prober-warning);
+        do {
+            assert_screen(\@tags);
+            handle_warning_install_os_prober() if match_has_tag('os-prober-warning');
+        } until(match_has_tag('yast2-kdump-restart-info'));
```

Comment:
you probably don't need such a infinite loop ... I think `assert_screen` already is a loop itself with some sleep waiting for tags...
I am thinking about something like this:
```
assert_screen(\@tags, 200);
if match_has_tag('os-prober-warning') {
    handle_warning_install_os_prober();
    assert_screen('yast2-kdump-restart-info', 200);
} 
```

Reply:
Actually the the loop should be faster. `assert_screen` has an infinite loop. when `assert_screen(\@tags, 200);` will has constant delay, where the `until` loop will exit as soon as it finds a match. 
`match_has_tag` has no sleep or anything.

Reply:
And @ggardet was right. Compare [last kdump_and_crash](https://openqa.opensuse.org/tests/1839626)  11m 26s with [previous](https://openqa.opensuse.org/tests/1838689) 12m 14s.

Reply:
@jlausuch sometimes a loop like this is necessary. Compare to https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/x11/kontact.pm#L39

`assert_screen` blocks until *any* tag matches but `do { … } until match_has_tag 'foo'` runs until the *specific* tag matches. Like the in the following example:

```perl
do {
    assert_screen ['expected-end-tag', 'random-popup1', 'random-popup2'];
    // do something about "random popups" when they happen
} until (match_has_tag 'expected-end-tag');
```

Reply:
Ok, you all convinced me :)
And thanks for the example, I struggled to find examples with these kind of loops.